src/pybind/mgr/cephadm/services/monitoring.py

```diff
@@ -256,86 +255,46 @@ def generate_config(
             self,
             daemon_spec: CephadmDaemonDeploySpec,
     ) -> Tuple[Dict[str, Any], List[str]]:
+
         assert self.TYPE == daemon_spec.daemon_type
         deps = []  # type: List[str]
-
-        # scrape mgrs
-        mgr_scrape_list = []
-        mgr_map = self.mgr.get('mgr_map')
         port = cast(int, self.mgr.get_module_option_ex(
             'prometheus', 'server_port', self.DEFAULT_MGR_PROMETHEUS_PORT))
         deps.append(str(port))
-        t = mgr_map.get('services', {}).get('prometheus', None)
+        # add an explicit dependency on the active manager. This will force to
+        # re-deploy prometheus if the mgr has changed (due to a fail-over i.e).
+        deps.append(self.mgr.get_active_mgr().name())
```

Comment:
yeah it was as far as I can remember, since I changed the code (and part of it has been moved to service_discovery) the way we calculate the deps has changed.

Reply:
@adk3798 now I remember why I changed it: in the old code I think both monitors (active and standby) were added as dependencies. In the new one only the active monitor is added as dependency to force a redeploy if mgr fails over.